fixham/internal/version/version.go

```diff
@@ -0,0 +1,79 @@
+package version
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+
+	"github.com/projectcalico/calico/fixham/internal/calico"
+)
+
+type Version string
+
+func (v *Version) Set(value string) error {
+	_v, err := semver.NewVersion(strings.TrimPrefix(value, "v"))
+	if err != nil {
+		return err
+	}
+	*v = Version(_v.String())
+	return nil
+}
+
+func (v *Version) String() string {
+	return string(*v)
+}
+
+func (v *Version) FormattedString() string {
+	if *v == "" {
```

Comment:
when is `v` going to be an empty string?

Reply:
TBH only from user error 

```go
v := Version()
v.FormattedString() // this will be an empty string
```